[GUIDES] Improve inference providers documentation with guides (#1797)
huggingface/hub-docs: docs/inference-providers/guides/building-first-app.md

    """Process uploaded audio file and return transcript + summary"""
    if audio_file is None:
        return "Please upload an audio file.", ""
    
    # We'll implement the AI logic next
    return "Transcript will appear here...", "Summary will appear here..."

# Create the Gradio interface
app = gr.Interface(
    fn=process_meeting_audio,
    inputs=gr.Audio(label="Upload Meeting Audio", type="filepath"),
    outputs=[
        gr.Textbox(label="Transcript", lines=10),
        gr.Textbox(label="Summary & Action Items", lines=8)
    ],
    title="🎤 AI Meeting Notes",
    description="Upload an audio file to get an instant transcript and summary with action items."
)

if __name__ == "__main__":
    app.launch()
```

Here we're using Gradio's `gr.Audio` component to either upload an audio file or use the microphone input. We're keeping things simple with two outputs: a transcript and a summary with action items.

## Step 3: Add Speech Transcription

Now let's implement the transcription using `fal.ai` and OpenAI's `whisper-large-v3` model for fast, reliable speech processing:

```python
def transcribe_audio(audio_file_path):
    """Transcribe audio using fal.ai for speed"""

>     client = InferenceClient(provider="fal-ai")

Reviewer: ```suggestion
    client = InferenceClient(provider="auto")
```

we could use `auto` here and essentially tell people that we route to the best and most performant provider for you (even easier for a beginner)

                const transcript = await transcribe(file);
                console.log('✅ Transcription completed:', transcript.substring(0, 100) + '...');
                
                console.log('🖊️ Starting summarization...');
                const summary = await summarize(transcript);
                console.log('✅ Summary completed:', summary.substring(0, 100) + '...');
                
                document.getElementById('transcript').textContent = transcript;
                document.getElementById('summary').textContent = summary;
                
                hide(document.getElementById('loading'));
                show(document.getElementById('results'));
            } catch (error) {
                hide(document.getElementById('loading'));
                showError(`Error: ${error.message}`);
            }
        };
        
        async function transcribe(file) {
            const client = new InferenceClient(HF_TOKEN);

            const output = await client.automaticSpeechRecognition({
                data: file,
                model: "openai/whisper-large-v3-turbo",
                provider: "auto"
            });
            

>             return output.text || output || 'Transcription completed';

Reviewer 2: (nit) 
```
return output.text || output || 'Transcription completed';
```

        document.getElementById('summary').textContent = summary;
        
        hide(document.getElementById('loading'));
        show(document.getElementById('results'));
    } catch (error) {
        hide(document.getElementById('loading'));
        showError(`Error: ${error.message}`);
    }
};
```

We'll also need to implement the `transcribe` and `summarize` functions.

</hfoption>
</hfoptions>

## Step 3: Add Speech Transcription

<hfoptions id="transcription">
<hfoption id="python">

Now let's implement the transcription using `fal.ai` and OpenAI's `whisper-large-v3` model for fast, reliable speech processing:

```python
def transcribe_audio(audio_file_path):
    """Transcribe audio using fal.ai for speed"""
    client = InferenceClient(provider="auto")
    
    # Pass the file path directly - the client handles file reading
    transcript = client.automatic_speech_recognition(
        audio=audio_file_path,
        model="openai/whisper-large-v3"
    )
    
    return transcript.text
```

Using the `auto` provider will automatically select the best provider for the model we're using.

</hfoption>
<hfoption id="javascript">

> We'll use the Hugging Face Inference client with automatic provider selection:

Reviewer 3: Let’s keep the same explanatory text for both python and js (for this step and the next one), only the code snippets should differ
```suggestion
Now let's implement the transcription using OpenAI's `whisper-large-v3` model for fast, reliable speech processing. 
<Tip>
We'll use the `auto` provider to automatically select the first available provider for the model. You can define your own priority list of providers in the [Inference Providers](https://huggingface.co/settings/inference-providers) page.
</Tip>
```

```javascript
import { InferenceClient } from 'https://esm.sh/@huggingface/inference';

async function transcribe(file) {
    const client = new InferenceClient(HF_TOKEN);

    const output = await client.automaticSpeechRecognition({
        data: file,
        model: "openai/whisper-large-v3-turbo",
        provider: "auto"
    });
    
    return output.text || output || 'Transcription completed';
}
```

> Using the `auto` provider will automatically select the best provider for the model we're using.

Reviewer 3: added a tip above 
```suggestion
```

</hfoption>
</hfoptions>

## Step 4: Add AI Summarization

<hfoptions id="summarization">
<hfoption id="python">

> Next, we'll use a powerful language model like `deepseek-ai/DeepSeek-R1-0528` from DeepSeek via an Inference Provider. 
> 
> <Tip>
> 
> We'll use the `auto` provider to automatically select the best provider for the model. You can define your own priority list of providers in the [Inference Providers](https://huggingface.co/settings/inference-providers) page.
> 
> </Tip>

Reviewer 3: no need to repeat the tip here.
```suggestion
Next, we'll use a powerful language model like `deepseek-ai/DeepSeek-R1-0528` from DeepSeek via an Inference Provider, and just like in the previous step, we'll use the `auto` provider to automatically select the first available provider for the model.
We will define a custom prompt to ensure the output is formatted as a summary with action items and decisions made:
```

```python
def generate_summary(transcript):
    """Generate summary using an Inference Provider"""
    client = InferenceClient(provider="auto")
    
    prompt = f"""
    Analyze this meeting transcript and provide:
    1. A concise summary of key points
    2. Action items with responsible parties
    3. Important decisions made
    
    Transcript: {transcript}
    
    Format with clear sections:
    ## Summary
    ## Action Items  
    ## Decisions Made
    """
    
    response = client.chat.completions.create(
        model="deepseek-ai/DeepSeek-R1-0528",
        messages=[{"role": "user", "content": prompt}],
        max_tokens=1000
    )
    
    return response.choices[0].message.content
```

> Note, we're also defining a custom summary prompt to ensure the output is formatted as a summary with action items and decisions made.

Reviewer 3: (see my comment above) added the note before the snippet 
```suggestion
```

</hfoption>
<hfoption id="javascript">

> We'll use the chat completion API with automatic provider selection again, and define a custom prompt to ensure the output is formatted as a summary with action items and decisions made:

Reviewer 3: keeping the explanatory text consistent between python and js
```suggestion
Next, we'll use a powerful language model like `deepseek-ai/DeepSeek-R1-0528` from DeepSeek via an Inference Provider, and just like in the previous step, we'll use the `auto` provider to automatically select the first available provider for the model.
We will define a custom prompt to ensure the output is formatted as a summary with action items and decisions made:
```

```javascript
async function summarize(transcript) {
    const client = new InferenceClient(HF_TOKEN);

    const prompt = `Analyze this meeting transcript and provide:
    1. A concise summary of key points
    2. Action items with responsible parties
    3. Important decisions made

    Transcript: ${transcript}

    Format with clear sections:
    ## Summary
    ## Action Items  
    ## Decisions Made`;

    const response = await client.chatCompletion({
        model: "deepseek-ai/DeepSeek-R1-0528",
        messages: [
            {
                role: "user", 
                content: prompt
            }
        ],
        max_tokens: 1000
    }, {
        provider: "auto"
    });
    
    return response.choices?.[0]?.message?.content || response || 'No summary available';
}
```

> We're using automatic provider selection which will choose the best available provider for the model.

Reviewer 3: ```suggestion
```

        return "Please upload an audio file.", ""

    try:
        # Step 1: Transcribe
        transcript = transcribe_audio(audio_file)

        # Step 2: Summarize
        summary = generate_summary(transcript)

        return transcript, summary

    except Exception as e:
        return f"Error processing audio: {str(e)}", ""

# Create Gradio interface
app = gr.Interface(
    fn=process_meeting_audio,
    inputs=gr.Audio(label="Upload Meeting Audio", type="filepath"),
    outputs=[
        gr.Textbox(label="Transcript", lines=10),
        gr.Textbox(label="Summary & Action Items", lines=8),
    ],
    title="🎤 AI Meeting Notes",
    description="Upload audio to get instant transcripts and summaries.",
)

if __name__ == "__main__":
    app.launch()
```

Our app will run on port 7860 and look like this:

![Gradio app](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers-guides/gradio-app.png)

</details>

To deploy, we'll need to create a new Space and upload our files.

1. **Create a new Space**: Go to [huggingface.co/new-space](https://huggingface.co/new-space)
2. **Choose Gradio SDK** and make it public

> 3. **Upload your files**: Upload `app.py` and `requirements.txt`

Reviewer 3: no need for a `requirements.txt` no?

> 3. **Upload your files**: Upload `app.py` and `requirements.txt`

Author: ```suggestion
3. **Upload your files**: Upload `app.py`
```
